batch/test/test_dag.py

```diff
@@ -141,32 +111,15 @@ def test_cancel_left_after_tail(client):
         assert node.status()['state'] == 'Cancelled'
 
 
-def test_delete(client):
-    batch = client.create_batch()
-    head = batch.create_job('alpine:3.8', command=['echo', 'head'])
-    left = batch.create_job('alpine:3.8', command=['echo', 'left'], parent_ids=[head.id])
-    right = batch.create_job('alpine:3.8', command=['echo', 'right'], parent_ids=[head.id])
-    tail = batch.create_job(
-        'alpine:3.8',
-        command=['/bin/sh', '-c', 'while true; do sleep 86000; done'],
-        parent_ids=[left.id, right.id])
-    tail.delete()
-    status = batch.wait()
-    assert batch_status_job_counter(status, 'Complete') >= 3
-    for node in [head, left, right]:
-        status = node.status()
-        assert status['state'] == 'Complete'
-        assert status['exit_code'] == 0
-
-
 def test_one_of_two_parent_ids_cancelled(client):
     batch = client.create_batch()
     left = batch.create_job(
         'alpine:3.8',
         command=['/bin/sh', '-c', 'while true; do sleep 86000; done'])
     right = batch.create_job('alpine:3.8', command=['echo', 'right'])
     tail = batch.create_job('alpine:3.8', command=['echo', 'tail'], parent_ids=[left.id, right.id])
-    left.cancel()
+    right.wait()
```

Comment:
By not waiting on left do you introduce a race condition? Conversely, do you need to wait on right?

Reply:
Namely in other tests we’re waiting on left, right, head I consistently. We should either make them consistent I think, or explicitly indicate we’re testing for race conditions.

Reply:
We don't want to wait on left here. It's going to sleep for a long time. The point of this test is to test what happens when left is cancelled but right is done already to make sure the child is also cancelled.

Reply:
this test no longer does that. You’re testing what happens when right is done, left is ? and the final parent (the batch) is cancelled.

Reply:
I'll just take out this test.